<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Pre-compiled GO Gene Sets • BioMartGOGeneSets</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Pre-compiled GO Gene Sets">
<meta property="og:description" content="BioMartGOGeneSets">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">BioMartGOGeneSets</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.99.4</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/biomart.html">Retrieve GO Gene Sets from BioMart</a>
    </li>
    <li>
      <a href="../articles/genesets.html">Pre-compiled GO Gene Sets</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/jokergoo/BioMartGOGeneSets/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Pre-compiled GO Gene Sets</h1>
                        <h4 data-toc-skip class="author">Zuguang Gu (<a href="mailto:z.gu@dkfz.de" class="email">z.gu@dkfz.de</a>)</h4>
            
            <h4 data-toc-skip class="date">2022-10-06</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/jokergoo/BioMartGOGeneSets/blob/HEAD/vignettes/genesets.Rmd" class="external-link"><code>vignettes/genesets.Rmd</code></a></small>
      <div class="hidden name"><code>genesets.Rmd</code></div>

    </div>

    
    
<p>The <strong>BioMartGOGeneSets</strong> contains pre-compiled GO gene sets for a huge number of organisms supported in <a href="https://www.ensembl.org/info/data/biomart/index.html" class="external-link">BioMart</a>. There are two types of data: 1. genes and 2 gene sets.</p>
<div class="section level2">
<h2 id="retrieve-genes">Retrieve genes<a class="anchor" aria-label="anchor" href="#retrieve-genes"></a>
</h2>
<p>To obtain the genes, use the function <code><a href="../reference/getBioMartGenes.html">getBioMartGenes()</a></code>. You need to provide a proper “dataset”, which can be found with the function <code><a href="../reference/supportedOrganisms.html">supportedOrganisms()</a></code> (A complete list can be also found from <a href="https://htmlpreview.github.io/?https://github.com/jokergoo/BioMartGOGeneSets/blob/master/inst/extdata/global_table.html" class="external-link uri">https://htmlpreview.github.io/?https://github.com/jokergoo/BioMartGOGeneSets/blob/master/inst/extdata/global_table.html</a>). Here we use the dataset <code>"hsapiens_gene_ensembl"</code> which is for human.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/jokergoo/BioMartGOGeneSets" class="external-link">BioMartGOGeneSets</a></span><span class="op">)</span></span>
<span><span class="va">gr</span> <span class="op">=</span> <span class="fu"><a href="../reference/getBioMartGenes.html">getBioMartGenes</a></span><span class="op">(</span><span class="st">"hsapiens_gene_ensembl"</span><span class="op">)</span></span>
<span><span class="va">gr</span></span></code></pre></div>
<pre><code><span><span class="co">## GRanges object with 68324 ranges and 4 metadata columns:</span></span>
<span><span class="co">##                   seqnames              ranges strand | ensembl_gene_id</span></span>
<span><span class="co">##                      &lt;Rle&gt;           &lt;IRanges&gt;  &lt;Rle&gt; |     &lt;character&gt;</span></span>
<span><span class="co">##   ENSG00000000003        X 100627108-100639991      - | ENSG00000000003</span></span>
<span><span class="co">##   ENSG00000000005        X 100584936-100599885      + | ENSG00000000005</span></span>
<span><span class="co">##   ENSG00000000419       20   50934867-50959140      - | ENSG00000000419</span></span>
<span><span class="co">##   ENSG00000000457        1 169849631-169894267      - | ENSG00000000457</span></span>
<span><span class="co">##   ENSG00000000460        1 169662007-169854080      + | ENSG00000000460</span></span>
<span><span class="co">##               ...      ...                 ...    ... .             ...</span></span>
<span><span class="co">##   ENSG00000290162        X 119807433-119807581      - | ENSG00000290162</span></span>
<span><span class="co">##   ENSG00000290163        X 119815164-119815307      - | ENSG00000290163</span></span>
<span><span class="co">##   ENSG00000290164        X 119822906-119823043      - | ENSG00000290164</span></span>
<span><span class="co">##   ENSG00000290165        X 119831616-119831759      - | ENSG00000290165</span></span>
<span><span class="co">##   ENSG00000290166       19     1163451-1165333      + | ENSG00000290166</span></span>
<span><span class="co">##                             gene_biotype   entrezgene_id external_gene_name</span></span>
<span><span class="co">##                              &lt;character&gt; &lt;CharacterList&gt;        &lt;character&gt;</span></span>
<span><span class="co">##   ENSG00000000003         protein_coding            7105             TSPAN6</span></span>
<span><span class="co">##   ENSG00000000005         protein_coding           64102               TNMD</span></span>
<span><span class="co">##   ENSG00000000419         protein_coding            8813               DPM1</span></span>
<span><span class="co">##   ENSG00000000457         protein_coding           57147              SCYL3</span></span>
<span><span class="co">##   ENSG00000000460         protein_coding           55732           C1orf112</span></span>
<span><span class="co">##               ...                    ...             ...                ...</span></span>
<span><span class="co">##   ENSG00000290162 unprocessed_pseudogene            &lt;NA&gt;               &lt;NA&gt;</span></span>
<span><span class="co">##   ENSG00000290163 unprocessed_pseudogene            &lt;NA&gt;               &lt;NA&gt;</span></span>
<span><span class="co">##   ENSG00000290164 unprocessed_pseudogene            &lt;NA&gt;               &lt;NA&gt;</span></span>
<span><span class="co">##   ENSG00000290165 unprocessed_pseudogene            &lt;NA&gt;               &lt;NA&gt;</span></span>
<span><span class="co">##   ENSG00000290166                 lncRNA            &lt;NA&gt;               &lt;NA&gt;</span></span>
<span><span class="co">##   -------</span></span>
<span><span class="co">##   seqinfo: 445 sequences from an unspecified genome; no seqlengths</span></span></code></pre>
<p>The returned value is a <code>GRanges</code> object which contains the coordinates of genes. The meta columns contain additional information of genes such as different gene IDs and type of genes.</p>
<p>You can also provide a “short name” for dataset and the function will perform a partial matching.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gr</span> <span class="op">=</span> <span class="fu"><a href="../reference/getBioMartGenes.html">getBioMartGenes</a></span><span class="op">(</span><span class="st">"hsapiens"</span><span class="op">)</span></span></code></pre></div>
<p>You can try the following command and see what will be printed:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gr</span> <span class="op">=</span> <span class="fu"><a href="../reference/getBioMartGenes.html">getBioMartGenes</a></span><span class="op">(</span><span class="st">"human"</span><span class="op">)</span></span></code></pre></div>
<p>Chromosome names from Ensembl have no “chr” prefix. You can set <code>add_chr_prefix = TURE</code> to add <code>"chr"</code> prefix to some of the chromosome names. Internally it uses <code>GenomeInfoDb::seqlevelsStyle(gr) = "UCSC"</code> to check and to add <code>"chr"</code> prefix.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gr</span> <span class="op">=</span> <span class="fu"><a href="../reference/getBioMartGenes.html">getBioMartGenes</a></span><span class="op">(</span><span class="st">"hsapiens_gene_ensembl"</span>, add_chr_prefix <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">gr</span></span></code></pre></div>
<pre><code><span><span class="co">## GRanges object with 68324 ranges and 4 metadata columns:</span></span>
<span><span class="co">##                   seqnames              ranges strand | ensembl_gene_id</span></span>
<span><span class="co">##                      &lt;Rle&gt;           &lt;IRanges&gt;  &lt;Rle&gt; |     &lt;character&gt;</span></span>
<span><span class="co">##   ENSG00000000003     chrX 100627108-100639991      - | ENSG00000000003</span></span>
<span><span class="co">##   ENSG00000000005     chrX 100584936-100599885      + | ENSG00000000005</span></span>
<span><span class="co">##   ENSG00000000419    chr20   50934867-50959140      - | ENSG00000000419</span></span>
<span><span class="co">##   ENSG00000000457     chr1 169849631-169894267      - | ENSG00000000457</span></span>
<span><span class="co">##   ENSG00000000460     chr1 169662007-169854080      + | ENSG00000000460</span></span>
<span><span class="co">##               ...      ...                 ...    ... .             ...</span></span>
<span><span class="co">##   ENSG00000290162     chrX 119807433-119807581      - | ENSG00000290162</span></span>
<span><span class="co">##   ENSG00000290163     chrX 119815164-119815307      - | ENSG00000290163</span></span>
<span><span class="co">##   ENSG00000290164     chrX 119822906-119823043      - | ENSG00000290164</span></span>
<span><span class="co">##   ENSG00000290165     chrX 119831616-119831759      - | ENSG00000290165</span></span>
<span><span class="co">##   ENSG00000290166    chr19     1163451-1165333      + | ENSG00000290166</span></span>
<span><span class="co">##                             gene_biotype   entrezgene_id external_gene_name</span></span>
<span><span class="co">##                              &lt;character&gt; &lt;CharacterList&gt;        &lt;character&gt;</span></span>
<span><span class="co">##   ENSG00000000003         protein_coding            7105             TSPAN6</span></span>
<span><span class="co">##   ENSG00000000005         protein_coding           64102               TNMD</span></span>
<span><span class="co">##   ENSG00000000419         protein_coding            8813               DPM1</span></span>
<span><span class="co">##   ENSG00000000457         protein_coding           57147              SCYL3</span></span>
<span><span class="co">##   ENSG00000000460         protein_coding           55732           C1orf112</span></span>
<span><span class="co">##               ...                    ...             ...                ...</span></span>
<span><span class="co">##   ENSG00000290162 unprocessed_pseudogene            &lt;NA&gt;               &lt;NA&gt;</span></span>
<span><span class="co">##   ENSG00000290163 unprocessed_pseudogene            &lt;NA&gt;               &lt;NA&gt;</span></span>
<span><span class="co">##   ENSG00000290164 unprocessed_pseudogene            &lt;NA&gt;               &lt;NA&gt;</span></span>
<span><span class="co">##   ENSG00000290165 unprocessed_pseudogene            &lt;NA&gt;               &lt;NA&gt;</span></span>
<span><span class="co">##   ENSG00000290166                 lncRNA            &lt;NA&gt;               &lt;NA&gt;</span></span>
<span><span class="co">##   -------</span></span>
<span><span class="co">##   seqinfo: 445 sequences from an unspecified genome; no seqlengths</span></span></code></pre>
<p>Note <code>add_chr_prefix</code> is just a helper argument. You can basically do the same as:</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gr</span> <span class="op">=</span> <span class="fu"><a href="../reference/getBioMartGenes.html">getBioMartGenes</a></span><span class="op">(</span><span class="st">"hsapiens_gene_ensembl"</span><span class="op">)</span></span>
<span><span class="fu">seqlevelsStyle</span><span class="op">(</span><span class="va">gr</span><span class="op">)</span> <span class="op">=</span> <span class="st">"UCSC"</span></span>
<span><span class="va">gr</span></span></code></pre></div>
<pre><code><span><span class="co">## GRanges object with 68324 ranges and 4 metadata columns:</span></span>
<span><span class="co">##                   seqnames              ranges strand | ensembl_gene_id</span></span>
<span><span class="co">##                      &lt;Rle&gt;           &lt;IRanges&gt;  &lt;Rle&gt; |     &lt;character&gt;</span></span>
<span><span class="co">##   ENSG00000000003     chrX 100627108-100639991      - | ENSG00000000003</span></span>
<span><span class="co">##   ENSG00000000005     chrX 100584936-100599885      + | ENSG00000000005</span></span>
<span><span class="co">##   ENSG00000000419    chr20   50934867-50959140      - | ENSG00000000419</span></span>
<span><span class="co">##   ENSG00000000457     chr1 169849631-169894267      - | ENSG00000000457</span></span>
<span><span class="co">##   ENSG00000000460     chr1 169662007-169854080      + | ENSG00000000460</span></span>
<span><span class="co">##               ...      ...                 ...    ... .             ...</span></span>
<span><span class="co">##   ENSG00000290162     chrX 119807433-119807581      - | ENSG00000290162</span></span>
<span><span class="co">##   ENSG00000290163     chrX 119815164-119815307      - | ENSG00000290163</span></span>
<span><span class="co">##   ENSG00000290164     chrX 119822906-119823043      - | ENSG00000290164</span></span>
<span><span class="co">##   ENSG00000290165     chrX 119831616-119831759      - | ENSG00000290165</span></span>
<span><span class="co">##   ENSG00000290166    chr19     1163451-1165333      + | ENSG00000290166</span></span>
<span><span class="co">##                             gene_biotype   entrezgene_id external_gene_name</span></span>
<span><span class="co">##                              &lt;character&gt; &lt;CharacterList&gt;        &lt;character&gt;</span></span>
<span><span class="co">##   ENSG00000000003         protein_coding            7105             TSPAN6</span></span>
<span><span class="co">##   ENSG00000000005         protein_coding           64102               TNMD</span></span>
<span><span class="co">##   ENSG00000000419         protein_coding            8813               DPM1</span></span>
<span><span class="co">##   ENSG00000000457         protein_coding           57147              SCYL3</span></span>
<span><span class="co">##   ENSG00000000460         protein_coding           55732           C1orf112</span></span>
<span><span class="co">##               ...                    ...             ...                ...</span></span>
<span><span class="co">##   ENSG00000290162 unprocessed_pseudogene            &lt;NA&gt;               &lt;NA&gt;</span></span>
<span><span class="co">##   ENSG00000290163 unprocessed_pseudogene            &lt;NA&gt;               &lt;NA&gt;</span></span>
<span><span class="co">##   ENSG00000290164 unprocessed_pseudogene            &lt;NA&gt;               &lt;NA&gt;</span></span>
<span><span class="co">##   ENSG00000290165 unprocessed_pseudogene            &lt;NA&gt;               &lt;NA&gt;</span></span>
<span><span class="co">##   ENSG00000290166                 lncRNA            &lt;NA&gt;               &lt;NA&gt;</span></span>
<span><span class="co">##   -------</span></span>
<span><span class="co">##   seqinfo: 445 sequences from an unspecified genome; no seqlengths</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="retrieve-gene-sets">Retrieve gene sets<a class="anchor" aria-label="anchor" href="#retrieve-gene-sets"></a>
</h2>
<p>To obtain the gene sets, use the function <code><a href="../reference/getBioMartGOGeneSets.html">getBioMartGOGeneSets()</a></code>. Also you need to provide the “dataset”. Here we use a different dataset: <code>"mmusculus_gene_ensembl"</code> (mouse).</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">lt</span> <span class="op">=</span> <span class="fu"><a href="../reference/getBioMartGOGeneSets.html">getBioMartGOGeneSets</a></span><span class="op">(</span><span class="st">"mmusculus_gene_ensembl"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">lt</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 16064</span></span></code></pre>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">lt</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">## $`GO:0000002`</span></span>
<span><span class="co">##  [1] "ENSMUSG00000019699" "ENSMUSG00000022615" "ENSMUSG00000028982"</span></span>
<span><span class="co">##  [4] "ENSMUSG00000039264" "ENSMUSG00000022889" "ENSMUSG00000030557"</span></span>
<span><span class="co">##  [7] "ENSMUSG00000030879" "ENSMUSG00000038084" "ENSMUSG00000032449"</span></span>
<span><span class="co">## [10] "ENSMUSG00000041064" "ENSMUSG00000033845" "ENSMUSG00000107283"</span></span>
<span><span class="co">## [13] "ENSMUSG00000027424" "ENSMUSG00000036875" "ENSMUSG00000038225"</span></span>
<span><span class="co">## [16] "ENSMUSG00000022292" "ENSMUSG00000025209" "ENSMUSG00000004069"</span></span>
<span><span class="co">## [19] "ENSMUSG00000039176" "ENSMUSG00000029911" "ENSMUSG00000020718"</span></span>
<span><span class="co">## [22] "ENSMUSG00000030978" "ENSMUSG00000026496" "ENSMUSG00000028893"</span></span>
<span><span class="co">## [25] "ENSMUSG00000002814" "ENSMUSG00000026365" "ENSMUSG00000015337"</span></span>
<span><span class="co">## [28] "ENSMUSG00000034203" "ENSMUSG00000059552" "ENSMUSG00000055660"</span></span>
<span><span class="co">## [31] "ENSMUSG00000028455" "ENSMUSG00000030314" "ENSMUSG00000020697"</span></span>
<span><span class="co">## [34] "ENSMUSG00000036923" "ENSMUSG00000032633" "ENSMUSG00000029167"</span></span></code></pre>
<p>The variable <code>lt</code> is a list of vectors where each vector corresponds to a GO gene set with Ensembl IDs as gene identifiers.</p>
<p>You can try the following command and see what will be printed:</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">lt</span> <span class="op">=</span> <span class="fu"><a href="../reference/getBioMartGOGeneSets.html">getBioMartGOGeneSets</a></span><span class="op">(</span><span class="st">"mouse"</span><span class="op">)</span></span></code></pre></div>
<p>In <code><a href="../reference/getBioMartGOGeneSets.html">getBioMartGOGeneSets()</a></code>, argument <code>as_table</code> can be set to <code>TRUE</code>, then the function returns a data frame.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tb</span> <span class="op">=</span> <span class="fu"><a href="../reference/getBioMartGOGeneSets.html">getBioMartGOGeneSets</a></span><span class="op">(</span><span class="st">"mmusculus_gene_ensembl"</span>, as_table <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">tb</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##   go_geneset       ensembl_gene</span></span>
<span><span class="co">## 1 GO:0000002 ENSMUSG00000019699</span></span>
<span><span class="co">## 2 GO:0000002 ENSMUSG00000022615</span></span>
<span><span class="co">## 3 GO:0000002 ENSMUSG00000028982</span></span>
<span><span class="co">## 4 GO:0000002 ENSMUSG00000039264</span></span>
<span><span class="co">## 5 GO:0000002 ENSMUSG00000022889</span></span>
<span><span class="co">## 6 GO:0000002 ENSMUSG00000030557</span></span></code></pre>
<p>Argument <code>ontology</code> controls which category of GO gene sets. Possible values should be <code>"BP"</code>, <code>"CC"</code> and <code>"MF"</code>.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/getBioMartGOGeneSets.html">getBioMartGOGeneSets</a></span><span class="op">(</span><span class="st">"mmusculus_gene_ensembl"</span>, ontology <span class="op">=</span> <span class="st">"BP"</span><span class="op">)</span> <span class="co"># the default one</span></span>
<span><span class="fu"><a href="../reference/getBioMartGOGeneSets.html">getBioMartGOGeneSets</a></span><span class="op">(</span><span class="st">"mmusculus_gene_ensembl"</span>, ontology <span class="op">=</span> <span class="st">"CC"</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/getBioMartGOGeneSets.html">getBioMartGOGeneSets</a></span><span class="op">(</span><span class="st">"mmusculus_gene_ensembl"</span>, ontology <span class="op">=</span> <span class="st">"MF"</span><span class="op">)</span></span></code></pre></div>
<p>Last, argument <code>gene_id_type</code> can be set to <code>"entrez_gene"</code> or <code>"gene_symbol"</code>, then genes in the gene sets are in Entrez IDs or gene symbols. Note this depends on specific organisms, that not every organism supports Entrez IDs or gene symbols.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">lt</span> <span class="op">=</span> <span class="fu"><a href="../reference/getBioMartGOGeneSets.html">getBioMartGOGeneSets</a></span><span class="op">(</span><span class="st">"mmusculus_gene_ensembl"</span>, gene_id_type <span class="op">=</span> <span class="st">"entrez_gene"</span><span class="op">)</span></span>
<span><span class="va">lt</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">## $`GO:0000002`</span></span>
<span><span class="co">##  [1] "23797"  "72962"  "70556"  "83408"  "27393"  "17258"  "27397"  "74143" </span></span>
<span><span class="co">##  [9] "192287" "208084" "27395"  "17527"  "74528"  "327762" "408022" "382985"</span></span>
<span><span class="co">## [17] "226153" "83945"  "18975"  "381760" "50776"  "20133"  "11545"  "230784"</span></span>
<span><span class="co">## [25] "21975"  "12628"  "13804"  "72170"  "22059"  "76781"  "66592"  "74244" </span></span>
<span><span class="co">## [33] "16882"  "216021" "216805" "19017"</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="version-of-the-data">Version of the data<a class="anchor" aria-label="anchor" href="#version-of-the-data"></a>
</h2>
<p>The object <code>BioMartGOGeneSets</code> contains the version and source of data.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">BioMartGOGeneSets</span></span></code></pre></div>
<pre><code><span><span class="co">## BioMart Gene Ontology gene sets</span></span>
<span><span class="co">##   Source: https://www.ensembl.org/info/data/biomart/index.html </span></span>
<span><span class="co">##   Number of organisms: 616 </span></span>
<span><span class="co">##   Marts: fungi_mart, genes_mart, metazoa_mart, plants_mart, protists_mart </span></span>
<span><span class="co">##   Built date: 2022-09-20, with biomaRt (2.52.0), GO.db (3.15.0)</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="session-info">Session Info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## R version 4.2.0 (2022-04-22)</span></span>
<span><span class="co">## Platform: x86_64-apple-darwin17.0 (64-bit)</span></span>
<span><span class="co">## Running under: macOS Big Sur/Monterey 10.16</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS:   /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRblas.0.dylib</span></span>
<span><span class="co">## LAPACK: /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRlapack.dylib</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">## [1] C/UTF-8/C/C/C/C</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span></span>
<span><span class="co">## [8] base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">## [1] GenomicRanges_1.48.0     GenomeInfoDb_1.32.2      IRanges_2.30.0          </span></span>
<span><span class="co">## [4] S4Vectors_0.34.0         BiocGenerics_0.42.0      BioMartGOGeneSets_0.99.4</span></span>
<span><span class="co">## [7] knitr_1.39              </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##  [1] XVector_0.36.0         magrittr_2.0.3         zlibbioc_1.42.0       </span></span>
<span><span class="co">##  [4] R6_2.5.1               ragg_1.2.2             rlang_1.0.4           </span></span>
<span><span class="co">##  [7] fastmap_1.1.0          stringr_1.4.0          tools_4.2.0           </span></span>
<span><span class="co">## [10] xfun_0.31              cli_3.3.0              jquerylib_0.1.4       </span></span>
<span><span class="co">## [13] htmltools_0.5.3        systemfonts_1.0.4      yaml_2.3.5            </span></span>
<span><span class="co">## [16] digest_0.6.29          rprojroot_2.0.3        pkgdown_2.0.6         </span></span>
<span><span class="co">## [19] textshaping_0.3.6      GenomeInfoDbData_1.2.8 purrr_0.3.4           </span></span>
<span><span class="co">## [22] bitops_1.0-7           sass_0.4.2             fs_1.5.2              </span></span>
<span><span class="co">## [25] RCurl_1.98-1.8         memoise_2.0.1          cachem_1.0.6          </span></span>
<span><span class="co">## [28] evaluate_0.15          rmarkdown_2.14         stringi_1.7.8         </span></span>
<span><span class="co">## [31] compiler_4.2.0         bslib_0.4.0            desc_1.4.1            </span></span>
<span><span class="co">## [34] jsonlite_1.8.0</span></span></code></pre>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Zuguang Gu.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
