<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Pre-compiled GO Gene Sets • BioMartGOGeneSets</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Pre-compiled GO Gene Sets">
<meta property="og:description" content="BioMartGOGeneSets">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">BioMartGOGeneSets</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.99.9</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/biomart.html">Retrieve GO Gene Sets from BioMart</a>
    </li>
    <li>
      <a href="../articles/genesets.html">Pre-compiled GO Gene Sets</a>
    </li>
    <li>
      <a href="../articles/supported_organisms.html">BioMart Gene Ontology Gene Sets Collections</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/jokergoo/BioMartGOGeneSets/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Pre-compiled GO Gene Sets</h1>
                        <h4 data-toc-skip class="author">Zuguang Gu (<a href="mailto:z.gu@dkfz.de" class="email">z.gu@dkfz.de</a>)</h4>
            
            <h4 data-toc-skip class="date">2023-02-12</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/jokergoo/BioMartGOGeneSets/blob/HEAD/vignettes/genesets.Rmd" class="external-link"><code>vignettes/genesets.Rmd</code></a></small>
      <div class="hidden name"><code>genesets.Rmd</code></div>

    </div>

    
    
<p>The <strong>BioMartGOGeneSets</strong> contains pre-compiled GO gene sets for a huge number of organisms supported in <a href="https://www.ensembl.org/info/data/biomart/index.html" class="external-link">BioMart</a>. There are two types of data: 1. genes and 2 gene sets.</p>
<div class="section level2">
<h2 id="retrieve-genes">Retrieve genes<a class="anchor" aria-label="anchor" href="#retrieve-genes"></a>
</h2>
<p>To obtain the genes, use the function <code><a href="../reference/getBioMartGenes.html">getBioMartGenes()</a></code>. You need to provide a proper “dataset”, which can be found with the function <code><a href="../reference/supportedOrganisms.html">supportedOrganisms()</a></code> (A complete list can be also found from “<a href="supported_organisms.html"><strong>BioMart Gene Ontology Gene Sets Collections</strong></a>)”. Here we use the dataset <code>"hsapiens_gene_ensembl"</code> as an example which is for human.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/jokergoo/BioMartGOGeneSets" class="external-link">BioMartGOGeneSets</a></span><span class="op">)</span></span>
<span><span class="va">gr</span> <span class="op">=</span> <span class="fu"><a href="../reference/getBioMartGenes.html">getBioMartGenes</a></span><span class="op">(</span><span class="st">"hsapiens_gene_ensembl"</span><span class="op">)</span></span>
<span><span class="va">gr</span></span></code></pre></div>
<pre><code><span><span class="co">## GRanges object with 69299 ranges and 4 metadata columns:</span></span>
<span><span class="co">##                   seqnames              ranges strand | ensembl_gene_id</span></span>
<span><span class="co">##                      &lt;Rle&gt;           &lt;IRanges&gt;  &lt;Rle&gt; |     &lt;character&gt;</span></span>
<span><span class="co">##   ENSG00000000003        X 100627108-100639991      - | ENSG00000000003</span></span>
<span><span class="co">##   ENSG00000000005        X 100584936-100599885      + | ENSG00000000005</span></span>
<span><span class="co">##   ENSG00000000419       20   50934867-50959140      - | ENSG00000000419</span></span>
<span><span class="co">##   ENSG00000000457        1 169849631-169894267      - | ENSG00000000457</span></span>
<span><span class="co">##   ENSG00000000460        1 169662007-169854080      + | ENSG00000000460</span></span>
<span><span class="co">##               ...      ...                 ...    ... .             ...</span></span>
<span><span class="co">##   ENSG00000291313       14 103334237-103335932      + | ENSG00000291313</span></span>
<span><span class="co">##   ENSG00000291314        X   10566888-10576955      - | ENSG00000291314</span></span>
<span><span class="co">##   ENSG00000291315        3   40312086-40312214      + | ENSG00000291315</span></span>
<span><span class="co">##   ENSG00000291316        8 144449582-144465430      - | ENSG00000291316</span></span>
<span><span class="co">##   ENSG00000291317        8 144463817-144465667      - | ENSG00000291317</span></span>
<span><span class="co">##                     gene_biotype   entrezgene_id external_gene_name</span></span>
<span><span class="co">##                      &lt;character&gt; &lt;CharacterList&gt;        &lt;character&gt;</span></span>
<span><span class="co">##   ENSG00000000003 protein_coding            7105             TSPAN6</span></span>
<span><span class="co">##   ENSG00000000005 protein_coding           64102               TNMD</span></span>
<span><span class="co">##   ENSG00000000419 protein_coding            8813               DPM1</span></span>
<span><span class="co">##   ENSG00000000457 protein_coding           57147              SCYL3</span></span>
<span><span class="co">##   ENSG00000000460 protein_coding           55732           C1orf112</span></span>
<span><span class="co">##               ...            ...             ...                ...</span></span>
<span><span class="co">##   ENSG00000291313 protein_coding            &lt;NA&gt;               &lt;NA&gt;</span></span>
<span><span class="co">##   ENSG00000291314 protein_coding            &lt;NA&gt;               &lt;NA&gt;</span></span>
<span><span class="co">##   ENSG00000291315 protein_coding            &lt;NA&gt;               &lt;NA&gt;</span></span>
<span><span class="co">##   ENSG00000291316 protein_coding          157542               &lt;NA&gt;</span></span>
<span><span class="co">##   ENSG00000291317 protein_coding           84773            TMEM276</span></span>
<span><span class="co">##   -------</span></span>
<span><span class="co">##   seqinfo: 445 sequences from an unspecified genome; no seqlengths</span></span></code></pre>
<p>The returned value is a <code>GRanges</code> object which contains the coordinates of genes. The meta columns contain additional information of genes such as different gene IDs and type of genes.</p>
<p>You can also provide a “short name” for dataset and the function will perform a partial matching.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gr</span> <span class="op">=</span> <span class="fu"><a href="../reference/getBioMartGenes.html">getBioMartGenes</a></span><span class="op">(</span><span class="st">"hsapiens"</span><span class="op">)</span></span></code></pre></div>
<p>You can try the following command and see what will be printed:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gr</span> <span class="op">=</span> <span class="fu"><a href="../reference/getBioMartGenes.html">getBioMartGenes</a></span><span class="op">(</span><span class="st">"human"</span><span class="op">)</span></span></code></pre></div>
<p>You can also provide the taxon id for the organism:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gr</span> <span class="op">=</span> <span class="fu"><a href="../reference/getBioMartGenes.html">getBioMartGenes</a></span><span class="op">(</span><span class="fl">9606</span><span class="op">)</span></span></code></pre></div>
<p>Chromosome names from Ensembl have no “chr” prefix. You can set <code>add_chr_prefix = TURE</code> to add <code>"chr"</code> prefix to some of the chromosome names. Internally it uses <code>GenomeInfoDb::seqlevelsStyle(gr) = "UCSC"</code> to check and to add <code>"chr"</code> prefix.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gr</span> <span class="op">=</span> <span class="fu"><a href="../reference/getBioMartGenes.html">getBioMartGenes</a></span><span class="op">(</span><span class="st">"hsapiens_gene_ensembl"</span>, add_chr_prefix <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">gr</span></span></code></pre></div>
<pre><code><span><span class="co">## GRanges object with 69299 ranges and 4 metadata columns:</span></span>
<span><span class="co">##                   seqnames              ranges strand | ensembl_gene_id</span></span>
<span><span class="co">##                      &lt;Rle&gt;           &lt;IRanges&gt;  &lt;Rle&gt; |     &lt;character&gt;</span></span>
<span><span class="co">##   ENSG00000000003     chrX 100627108-100639991      - | ENSG00000000003</span></span>
<span><span class="co">##   ENSG00000000005     chrX 100584936-100599885      + | ENSG00000000005</span></span>
<span><span class="co">##   ENSG00000000419    chr20   50934867-50959140      - | ENSG00000000419</span></span>
<span><span class="co">##   ENSG00000000457     chr1 169849631-169894267      - | ENSG00000000457</span></span>
<span><span class="co">##   ENSG00000000460     chr1 169662007-169854080      + | ENSG00000000460</span></span>
<span><span class="co">##               ...      ...                 ...    ... .             ...</span></span>
<span><span class="co">##   ENSG00000291313    chr14 103334237-103335932      + | ENSG00000291313</span></span>
<span><span class="co">##   ENSG00000291314     chrX   10566888-10576955      - | ENSG00000291314</span></span>
<span><span class="co">##   ENSG00000291315     chr3   40312086-40312214      + | ENSG00000291315</span></span>
<span><span class="co">##   ENSG00000291316     chr8 144449582-144465430      - | ENSG00000291316</span></span>
<span><span class="co">##   ENSG00000291317     chr8 144463817-144465667      - | ENSG00000291317</span></span>
<span><span class="co">##                     gene_biotype   entrezgene_id external_gene_name</span></span>
<span><span class="co">##                      &lt;character&gt; &lt;CharacterList&gt;        &lt;character&gt;</span></span>
<span><span class="co">##   ENSG00000000003 protein_coding            7105             TSPAN6</span></span>
<span><span class="co">##   ENSG00000000005 protein_coding           64102               TNMD</span></span>
<span><span class="co">##   ENSG00000000419 protein_coding            8813               DPM1</span></span>
<span><span class="co">##   ENSG00000000457 protein_coding           57147              SCYL3</span></span>
<span><span class="co">##   ENSG00000000460 protein_coding           55732           C1orf112</span></span>
<span><span class="co">##               ...            ...             ...                ...</span></span>
<span><span class="co">##   ENSG00000291313 protein_coding            &lt;NA&gt;               &lt;NA&gt;</span></span>
<span><span class="co">##   ENSG00000291314 protein_coding            &lt;NA&gt;               &lt;NA&gt;</span></span>
<span><span class="co">##   ENSG00000291315 protein_coding            &lt;NA&gt;               &lt;NA&gt;</span></span>
<span><span class="co">##   ENSG00000291316 protein_coding          157542               &lt;NA&gt;</span></span>
<span><span class="co">##   ENSG00000291317 protein_coding           84773            TMEM276</span></span>
<span><span class="co">##   -------</span></span>
<span><span class="co">##   seqinfo: 445 sequences from an unspecified genome; no seqlengths</span></span></code></pre>
<p>Note <code>add_chr_prefix</code> is just a helper argument. You can basically do the same as:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gr</span> <span class="op">=</span> <span class="fu"><a href="../reference/getBioMartGenes.html">getBioMartGenes</a></span><span class="op">(</span><span class="st">"hsapiens_gene_ensembl"</span><span class="op">)</span></span>
<span><span class="fu">GenomeInfoDb</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqlevelsStyle.html" class="external-link">seqlevelsStyle</a></span><span class="op">(</span><span class="va">gr</span><span class="op">)</span> <span class="op">=</span> <span class="st">"UCSC"</span></span>
<span><span class="va">gr</span></span></code></pre></div>
<pre><code><span><span class="co">## GRanges object with 69299 ranges and 4 metadata columns:</span></span>
<span><span class="co">##                   seqnames              ranges strand | ensembl_gene_id</span></span>
<span><span class="co">##                      &lt;Rle&gt;           &lt;IRanges&gt;  &lt;Rle&gt; |     &lt;character&gt;</span></span>
<span><span class="co">##   ENSG00000000003     chrX 100627108-100639991      - | ENSG00000000003</span></span>
<span><span class="co">##   ENSG00000000005     chrX 100584936-100599885      + | ENSG00000000005</span></span>
<span><span class="co">##   ENSG00000000419    chr20   50934867-50959140      - | ENSG00000000419</span></span>
<span><span class="co">##   ENSG00000000457     chr1 169849631-169894267      - | ENSG00000000457</span></span>
<span><span class="co">##   ENSG00000000460     chr1 169662007-169854080      + | ENSG00000000460</span></span>
<span><span class="co">##               ...      ...                 ...    ... .             ...</span></span>
<span><span class="co">##   ENSG00000291313    chr14 103334237-103335932      + | ENSG00000291313</span></span>
<span><span class="co">##   ENSG00000291314     chrX   10566888-10576955      - | ENSG00000291314</span></span>
<span><span class="co">##   ENSG00000291315     chr3   40312086-40312214      + | ENSG00000291315</span></span>
<span><span class="co">##   ENSG00000291316     chr8 144449582-144465430      - | ENSG00000291316</span></span>
<span><span class="co">##   ENSG00000291317     chr8 144463817-144465667      - | ENSG00000291317</span></span>
<span><span class="co">##                     gene_biotype   entrezgene_id external_gene_name</span></span>
<span><span class="co">##                      &lt;character&gt; &lt;CharacterList&gt;        &lt;character&gt;</span></span>
<span><span class="co">##   ENSG00000000003 protein_coding            7105             TSPAN6</span></span>
<span><span class="co">##   ENSG00000000005 protein_coding           64102               TNMD</span></span>
<span><span class="co">##   ENSG00000000419 protein_coding            8813               DPM1</span></span>
<span><span class="co">##   ENSG00000000457 protein_coding           57147              SCYL3</span></span>
<span><span class="co">##   ENSG00000000460 protein_coding           55732           C1orf112</span></span>
<span><span class="co">##               ...            ...             ...                ...</span></span>
<span><span class="co">##   ENSG00000291313 protein_coding            &lt;NA&gt;               &lt;NA&gt;</span></span>
<span><span class="co">##   ENSG00000291314 protein_coding            &lt;NA&gt;               &lt;NA&gt;</span></span>
<span><span class="co">##   ENSG00000291315 protein_coding            &lt;NA&gt;               &lt;NA&gt;</span></span>
<span><span class="co">##   ENSG00000291316 protein_coding          157542               &lt;NA&gt;</span></span>
<span><span class="co">##   ENSG00000291317 protein_coding           84773            TMEM276</span></span>
<span><span class="co">##   -------</span></span>
<span><span class="co">##   seqinfo: 445 sequences from an unspecified genome; no seqlengths</span></span></code></pre>
<p>For some not-well-studied organisms, there might be no “official chromosome name”. For example, <code>"cporcellus_gene_ensembl"</code> for the guinea Pig:</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gr</span> <span class="op">=</span> <span class="fu"><a href="../reference/getBioMartGenes.html">getBioMartGenes</a></span><span class="op">(</span><span class="st">"cporcellus_gene_ensembl"</span><span class="op">)</span></span>
<span><span class="va">gr</span></span></code></pre></div>
<pre><code><span><span class="co">## GRanges object with 26855 ranges and 4 metadata columns:</span></span>
<span><span class="co">##                        seqnames            ranges strand |    ensembl_gene_id</span></span>
<span><span class="co">##                           &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; |        &lt;character&gt;</span></span>
<span><span class="co">##   ENSCPOG00000000007 DS562868.1 22719983-22745621      - | ENSCPOG00000000007</span></span>
<span><span class="co">##   ENSCPOG00000000011 DS562863.1 58475996-58484540      + | ENSCPOG00000000011</span></span>
<span><span class="co">##   ENSCPOG00000000012 DS562864.1 51072990-51073943      + | ENSCPOG00000000012</span></span>
<span><span class="co">##   ENSCPOG00000000013 DS563093.1     409300-421802      + | ENSCPOG00000000013</span></span>
<span><span class="co">##   ENSCPOG00000000014 DS562923.1   9324624-9325632      + | ENSCPOG00000000014</span></span>
<span><span class="co">##                  ...        ...               ...    ... .                ...</span></span>
<span><span class="co">##   ENSCPOG00000040808 DS562855.1     547865-551436      + | ENSCPOG00000040808</span></span>
<span><span class="co">##   ENSCPOG00000040809 DS562862.1 50369714-50415125      - | ENSCPOG00000040809</span></span>
<span><span class="co">##   ENSCPOG00000040810 DS562864.1 43092199-43093149      - | ENSCPOG00000040810</span></span>
<span><span class="co">##   ENSCPOG00000040811 DS562863.1 28315518-28533596      + | ENSCPOG00000040811</span></span>
<span><span class="co">##   ENSCPOG00000040812 DS562897.1 10154287-10157556      - | ENSCPOG00000040812</span></span>
<span><span class="co">##                        gene_biotype   entrezgene_id external_gene_name</span></span>
<span><span class="co">##                         &lt;character&gt; &lt;CharacterList&gt;        &lt;character&gt;</span></span>
<span><span class="co">##   ENSCPOG00000000007 protein_coding       100730010             D2HGDH</span></span>
<span><span class="co">##   ENSCPOG00000000011 protein_coding       100730666              OTOL1</span></span>
<span><span class="co">##   ENSCPOG00000000012 protein_coding       100720088               &lt;NA&gt;</span></span>
<span><span class="co">##   ENSCPOG00000000013 protein_coding            &lt;NA&gt;             TRIM10</span></span>
<span><span class="co">##   ENSCPOG00000000014 protein_coding            &lt;NA&gt;               &lt;NA&gt;</span></span>
<span><span class="co">##                  ...            ...             ...                ...</span></span>
<span><span class="co">##   ENSCPOG00000040808 protein_coding       100735988             TMEM74</span></span>
<span><span class="co">##   ENSCPOG00000040809 protein_coding            &lt;NA&gt;               &lt;NA&gt;</span></span>
<span><span class="co">##   ENSCPOG00000040810     pseudogene            &lt;NA&gt;               &lt;NA&gt;</span></span>
<span><span class="co">##   ENSCPOG00000040811 protein_coding            &lt;NA&gt;               &lt;NA&gt;</span></span>
<span><span class="co">##   ENSCPOG00000040812        lincRNA            &lt;NA&gt;               &lt;NA&gt;</span></span>
<span><span class="co">##   -------</span></span>
<span><span class="co">##   seqinfo: 449 sequences from an unspecified genome; no seqlengths</span></span></code></pre>
<p>The sequence names are in a special format of <code>DS\d+</code>. The source of the format can be obtained by <code><a href="../reference/getBioMartGenomeInfo.html">getBioMartGenomeInfo()</a></code>. In the <code>seqname_style</code> element of the returned list, there are several examples that you can compare to.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/getBioMartGenomeInfo.html">getBioMartGenomeInfo</a></span><span class="op">(</span><span class="st">"cporcellus_gene_ensembl"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## $dataset</span></span>
<span><span class="co">## [1] "cporcellus_gene_ensembl"</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $version</span></span>
<span><span class="co">## [1] "Cavpor3.0"</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $name</span></span>
<span><span class="co">## [1] "Cavia porcellus (Guinea Pig)"</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $taxon_id</span></span>
<span><span class="co">## [1] "10141"</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $genbank_accession</span></span>
<span><span class="co">## [1] "GCA_000151735.1"</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $mart</span></span>
<span><span class="co">## [1] "genes_mart"</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $seqname_style</span></span>
<span><span class="co">## $seqname_style$`Sequence-Name`</span></span>
<span><span class="co">## [1] "supercont2_0" "supercont2_1" "supercont2_2" "supercont2_3" "supercont2_4"</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $seqname_style$`GenBank-Accn`</span></span>
<span><span class="co">## [1] "DS562855.1" "DS562856.1" "DS562857.1" "DS562858.1" "DS562859.1"</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $seqname_style$`RefSeq-Accn`</span></span>
<span><span class="co">## [1] "NT_176419.1" "NT_176418.1" "NT_176417.1" "NT_176416.1" "NT_176415.1"</span></span></code></pre>
<p>Now we know they are the GenBank accession IDs. Next we might want to change them to the <code>"Sequence-Name"</code> style. Simply use <code><a href="../reference/changeSeqnameStyle.html">changeSeqnameStyle()</a></code>.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gr2</span> <span class="op">=</span> <span class="fu"><a href="../reference/changeSeqnameStyle.html">changeSeqnameStyle</a></span><span class="op">(</span><span class="va">gr</span>, <span class="st">"cporcellus_gene_ensembl"</span>, </span>
<span>    seqname_style_from <span class="op">=</span> <span class="st">"GenBank-Accn"</span>, </span>
<span>    seqname_style_to <span class="op">=</span> <span class="st">"Sequence-Name"</span><span class="op">)</span></span>
<span><span class="va">gr2</span></span></code></pre></div>
<pre><code><span><span class="co">## GRanges object with 26818 ranges and 5 metadata columns:</span></span>
<span><span class="co">##                            seqnames            ranges strand |</span></span>
<span><span class="co">##                               &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; |</span></span>
<span><span class="co">##   ENSCPOG00000000007  supercont2_13 22719983-22745621      - |</span></span>
<span><span class="co">##   ENSCPOG00000000011   supercont2_8 58475996-58484540      + |</span></span>
<span><span class="co">##   ENSCPOG00000000012   supercont2_9 51072990-51073943      + |</span></span>
<span><span class="co">##   ENSCPOG00000000013 supercont2_238     409300-421802      + |</span></span>
<span><span class="co">##   ENSCPOG00000000014  supercont2_68   9324624-9325632      + |</span></span>
<span><span class="co">##                  ...            ...               ...    ... .</span></span>
<span><span class="co">##   ENSCPOG00000040808   supercont2_0     547865-551436      + |</span></span>
<span><span class="co">##   ENSCPOG00000040809   supercont2_7 50369714-50415125      - |</span></span>
<span><span class="co">##   ENSCPOG00000040810   supercont2_9 43092199-43093149      - |</span></span>
<span><span class="co">##   ENSCPOG00000040811   supercont2_8 28315518-28533596      + |</span></span>
<span><span class="co">##   ENSCPOG00000040812  supercont2_42 10154287-10157556      - |</span></span>
<span><span class="co">##                         ensembl_gene_id   gene_biotype   entrezgene_id</span></span>
<span><span class="co">##                             &lt;character&gt;    &lt;character&gt; &lt;CharacterList&gt;</span></span>
<span><span class="co">##   ENSCPOG00000000007 ENSCPOG00000000007 protein_coding       100730010</span></span>
<span><span class="co">##   ENSCPOG00000000011 ENSCPOG00000000011 protein_coding       100730666</span></span>
<span><span class="co">##   ENSCPOG00000000012 ENSCPOG00000000012 protein_coding       100720088</span></span>
<span><span class="co">##   ENSCPOG00000000013 ENSCPOG00000000013 protein_coding            &lt;NA&gt;</span></span>
<span><span class="co">##   ENSCPOG00000000014 ENSCPOG00000000014 protein_coding            &lt;NA&gt;</span></span>
<span><span class="co">##                  ...                ...            ...             ...</span></span>
<span><span class="co">##   ENSCPOG00000040808 ENSCPOG00000040808 protein_coding       100735988</span></span>
<span><span class="co">##   ENSCPOG00000040809 ENSCPOG00000040809 protein_coding            &lt;NA&gt;</span></span>
<span><span class="co">##   ENSCPOG00000040810 ENSCPOG00000040810     pseudogene            &lt;NA&gt;</span></span>
<span><span class="co">##   ENSCPOG00000040811 ENSCPOG00000040811 protein_coding            &lt;NA&gt;</span></span>
<span><span class="co">##   ENSCPOG00000040812 ENSCPOG00000040812        lincRNA            &lt;NA&gt;</span></span>
<span><span class="co">##                      external_gene_name .original_seqname</span></span>
<span><span class="co">##                             &lt;character&gt;       &lt;character&gt;</span></span>
<span><span class="co">##   ENSCPOG00000000007             D2HGDH        DS562868.1</span></span>
<span><span class="co">##   ENSCPOG00000000011              OTOL1        DS562863.1</span></span>
<span><span class="co">##   ENSCPOG00000000012               &lt;NA&gt;        DS562864.1</span></span>
<span><span class="co">##   ENSCPOG00000000013             TRIM10        DS563093.1</span></span>
<span><span class="co">##   ENSCPOG00000000014               &lt;NA&gt;        DS562923.1</span></span>
<span><span class="co">##                  ...                ...               ...</span></span>
<span><span class="co">##   ENSCPOG00000040808             TMEM74        DS562855.1</span></span>
<span><span class="co">##   ENSCPOG00000040809               &lt;NA&gt;        DS562862.1</span></span>
<span><span class="co">##   ENSCPOG00000040810               &lt;NA&gt;        DS562864.1</span></span>
<span><span class="co">##   ENSCPOG00000040811               &lt;NA&gt;        DS562863.1</span></span>
<span><span class="co">##   ENSCPOG00000040812               &lt;NA&gt;        DS562897.1</span></span>
<span><span class="co">##   -------</span></span>
<span><span class="co">##   seqinfo: 448 sequences from an unspecified genome; no seqlengths</span></span></code></pre>
<p>Sometimes the internal sequence names need to be reformatted to fit the input <code>gr</code>. In the second example ( <code>"apercula_gene_ensembl"</code> for the orange clownfish), the sequence names are in format of <code>1, 2, 3, ...</code>, while internally they are represented as <code>chr1, chr2, ...</code>.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gr</span> <span class="op">=</span> <span class="fu"><a href="../reference/getBioMartGenes.html">getBioMartGenes</a></span><span class="op">(</span><span class="st">"apercula_gene_ensembl"</span><span class="op">)</span></span>
<span><span class="va">gr</span></span></code></pre></div>
<pre><code><span><span class="co">## GRanges object with 24840 ranges and 3 metadata columns:</span></span>
<span><span class="co">##                      seqnames            ranges strand |    ensembl_gene_id</span></span>
<span><span class="co">##                         &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; |        &lt;character&gt;</span></span>
<span><span class="co">##   ENSAPEG00000000002        1       24225-56222      - | ENSAPEG00000000002</span></span>
<span><span class="co">##   ENSAPEG00000000003        1       84299-92425      - | ENSAPEG00000000003</span></span>
<span><span class="co">##   ENSAPEG00000000004        1     128917-153166      - | ENSAPEG00000000004</span></span>
<span><span class="co">##   ENSAPEG00000000005        1     159517-165941      + | ENSAPEG00000000005</span></span>
<span><span class="co">##   ENSAPEG00000000006        1     169070-172981      - | ENSAPEG00000000006</span></span>
<span><span class="co">##                  ...      ...               ...    ... .                ...</span></span>
<span><span class="co">##   ENSAPEG00000024837        1 42572675-42577642      + | ENSAPEG00000024837</span></span>
<span><span class="co">##   ENSAPEG00000024838        1 42576978-42651838      - | ENSAPEG00000024838</span></span>
<span><span class="co">##   ENSAPEG00000024839        1 42687220-42701205      - | ENSAPEG00000024839</span></span>
<span><span class="co">##   ENSAPEG00000024840        1 42704049-42726377      - | ENSAPEG00000024840</span></span>
<span><span class="co">##   ENSAPEG00000024841        1 42727223-42735871      - | ENSAPEG00000024841</span></span>
<span><span class="co">##                        gene_biotype external_gene_name</span></span>
<span><span class="co">##                         &lt;character&gt;        &lt;character&gt;</span></span>
<span><span class="co">##   ENSAPEG00000000002 protein_coding           emilin1b</span></span>
<span><span class="co">##   ENSAPEG00000000003 protein_coding               &lt;NA&gt;</span></span>
<span><span class="co">##   ENSAPEG00000000004 protein_coding               &lt;NA&gt;</span></span>
<span><span class="co">##   ENSAPEG00000000005 protein_coding               &lt;NA&gt;</span></span>
<span><span class="co">##   ENSAPEG00000000006 protein_coding              ctrb1</span></span>
<span><span class="co">##                  ...            ...                ...</span></span>
<span><span class="co">##   ENSAPEG00000024837 protein_coding                dut</span></span>
<span><span class="co">##   ENSAPEG00000024838 protein_coding               FBN1</span></span>
<span><span class="co">##   ENSAPEG00000024839 protein_coding               &lt;NA&gt;</span></span>
<span><span class="co">##   ENSAPEG00000024840 protein_coding          SECISBP2L</span></span>
<span><span class="co">##   ENSAPEG00000024841 protein_coding              cops2</span></span>
<span><span class="co">##   -------</span></span>
<span><span class="co">##   seqinfo: 72 sequences from an unspecified genome; no seqlengths</span></span></code></pre>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/getBioMartGenomeInfo.html">getBioMartGenomeInfo</a></span><span class="op">(</span><span class="st">"apercula_gene_ensembl"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## $dataset</span></span>
<span><span class="co">## [1] "apercula_gene_ensembl"</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $version</span></span>
<span><span class="co">## [1] "Nemo_v1"</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $name</span></span>
<span><span class="co">## [1] "Amphiprion percula (Orange clownfish)"</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $taxon_id</span></span>
<span><span class="co">## [1] "161767"</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $genbank_accession</span></span>
<span><span class="co">## [1] "GCA_003047355.1"</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $mart</span></span>
<span><span class="co">## [1] "genes_mart"</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $seqname_style</span></span>
<span><span class="co">## $seqname_style$`Sequence-Name`</span></span>
<span><span class="co">## [1] "chr1" "chr2" "chr3" "chr4" "chr5"</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $seqname_style$`GenBank-Accn`</span></span>
<span><span class="co">## [1] "CM009708.1" "CM009709.1" "CM009710.1" "CM009711.1" "CM009712.1"</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $seqname_style$`RefSeq-Accn`</span></span>
<span><span class="co">## [1] "na" "na" "na" "na" "na"</span></span></code></pre>
<p>In this case, we need to set the argument <code>reformat_from</code> as a function to reformat the internal format to fit the sequence names in <code>gr</code>. Also you can set <code>reformat_to</code> as a function to reformat the converted sequence names.</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gr2</span> <span class="op">=</span> <span class="fu"><a href="../reference/changeSeqnameStyle.html">changeSeqnameStyle</a></span><span class="op">(</span><span class="va">gr</span>, <span class="st">"apercula_gene_ensembl"</span>, </span>
<span>    seqname_style_from <span class="op">=</span> <span class="st">"Sequence-Name"</span>, </span>
<span>    seqname_style_to <span class="op">=</span> <span class="st">"GenBank-Accn"</span>, </span>
<span>    reformat_from <span class="op">=</span><span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">"chr"</span>, <span class="st">""</span>, <span class="va">x</span><span class="op">)</span>,</span>
<span>    reformat_to <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">"\\.\\d+$"</span>, <span class="st">""</span>, <span class="va">x</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span><span class="va">gr2</span></span></code></pre></div>
<pre><code><span><span class="co">## GRanges object with 24678 ranges and 4 metadata columns:</span></span>
<span><span class="co">##                      seqnames            ranges strand |    ensembl_gene_id</span></span>
<span><span class="co">##                         &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; |        &lt;character&gt;</span></span>
<span><span class="co">##   ENSAPEG00000000002 CM009708       24225-56222      - | ENSAPEG00000000002</span></span>
<span><span class="co">##   ENSAPEG00000000003 CM009708       84299-92425      - | ENSAPEG00000000003</span></span>
<span><span class="co">##   ENSAPEG00000000004 CM009708     128917-153166      - | ENSAPEG00000000004</span></span>
<span><span class="co">##   ENSAPEG00000000005 CM009708     159517-165941      + | ENSAPEG00000000005</span></span>
<span><span class="co">##   ENSAPEG00000000006 CM009708     169070-172981      - | ENSAPEG00000000006</span></span>
<span><span class="co">##                  ...      ...               ...    ... .                ...</span></span>
<span><span class="co">##   ENSAPEG00000024837 CM009708 42572675-42577642      + | ENSAPEG00000024837</span></span>
<span><span class="co">##   ENSAPEG00000024838 CM009708 42576978-42651838      - | ENSAPEG00000024838</span></span>
<span><span class="co">##   ENSAPEG00000024839 CM009708 42687220-42701205      - | ENSAPEG00000024839</span></span>
<span><span class="co">##   ENSAPEG00000024840 CM009708 42704049-42726377      - | ENSAPEG00000024840</span></span>
<span><span class="co">##   ENSAPEG00000024841 CM009708 42727223-42735871      - | ENSAPEG00000024841</span></span>
<span><span class="co">##                        gene_biotype external_gene_name .original_seqname</span></span>
<span><span class="co">##                         &lt;character&gt;        &lt;character&gt;       &lt;character&gt;</span></span>
<span><span class="co">##   ENSAPEG00000000002 protein_coding           emilin1b                 1</span></span>
<span><span class="co">##   ENSAPEG00000000003 protein_coding               &lt;NA&gt;                 1</span></span>
<span><span class="co">##   ENSAPEG00000000004 protein_coding               &lt;NA&gt;                 1</span></span>
<span><span class="co">##   ENSAPEG00000000005 protein_coding               &lt;NA&gt;                 1</span></span>
<span><span class="co">##   ENSAPEG00000000006 protein_coding              ctrb1                 1</span></span>
<span><span class="co">##                  ...            ...                ...               ...</span></span>
<span><span class="co">##   ENSAPEG00000024837 protein_coding                dut                 1</span></span>
<span><span class="co">##   ENSAPEG00000024838 protein_coding               FBN1                 1</span></span>
<span><span class="co">##   ENSAPEG00000024839 protein_coding               &lt;NA&gt;                 1</span></span>
<span><span class="co">##   ENSAPEG00000024840 protein_coding          SECISBP2L                 1</span></span>
<span><span class="co">##   ENSAPEG00000024841 protein_coding              cops2                 1</span></span>
<span><span class="co">##   -------</span></span>
<span><span class="co">##   seqinfo: 24 sequences from an unspecified genome; no seqlengths</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="retrieve-gene-sets">Retrieve gene sets<a class="anchor" aria-label="anchor" href="#retrieve-gene-sets"></a>
</h2>
<p>To obtain the gene sets, use the function <code><a href="../reference/getBioMartGOGeneSets.html">getBioMartGOGeneSets()</a></code>. Also you need to provide the “dataset”. Here we use a different dataset: <code>"mmusculus_gene_ensembl"</code> (mouse).</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">lt</span> <span class="op">=</span> <span class="fu"><a href="../reference/getBioMartGOGeneSets.html">getBioMartGOGeneSets</a></span><span class="op">(</span><span class="st">"mmusculus_gene_ensembl"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">lt</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 16060</span></span></code></pre>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">lt</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">## $`GO:0000002`</span></span>
<span><span class="co">##  [1] "ENSMUSG00000019699" "ENSMUSG00000028982" "ENSMUSG00000022615"</span></span>
<span><span class="co">##  [4] "ENSMUSG00000039264" "ENSMUSG00000030557" "ENSMUSG00000030879"</span></span>
<span><span class="co">##  [7] "ENSMUSG00000032449" "ENSMUSG00000038084" "ENSMUSG00000022889"</span></span>
<span><span class="co">## [10] "ENSMUSG00000041064" "ENSMUSG00000033845" "ENSMUSG00000107283"</span></span>
<span><span class="co">## [13] "ENSMUSG00000027424" "ENSMUSG00000036875" "ENSMUSG00000038225"</span></span>
<span><span class="co">## [16] "ENSMUSG00000022292" "ENSMUSG00000004069" "ENSMUSG00000039176"</span></span>
<span><span class="co">## [19] "ENSMUSG00000025209" "ENSMUSG00000029911" "ENSMUSG00000020718"</span></span>
<span><span class="co">## [22] "ENSMUSG00000030978" "ENSMUSG00000026496" "ENSMUSG00000028893"</span></span>
<span><span class="co">## [25] "ENSMUSG00000002814" "ENSMUSG00000026365" "ENSMUSG00000015337"</span></span>
<span><span class="co">## [28] "ENSMUSG00000034203" "ENSMUSG00000059552" "ENSMUSG00000055660"</span></span>
<span><span class="co">## [31] "ENSMUSG00000028455" "ENSMUSG00000030314" "ENSMUSG00000020697"</span></span>
<span><span class="co">## [34] "ENSMUSG00000036923" "ENSMUSG00000032633" "ENSMUSG00000029167"</span></span></code></pre>
<p>The variable <code>lt</code> is a list of vectors where each vector corresponds to a GO gene set with Ensembl IDs as gene identifiers.</p>
<p>You can try the following command and see what will be printed:</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">lt</span> <span class="op">=</span> <span class="fu"><a href="../reference/getBioMartGOGeneSets.html">getBioMartGOGeneSets</a></span><span class="op">(</span><span class="st">"mouse"</span><span class="op">)</span></span></code></pre></div>
<p>Remember you can also set the taxon ID:</p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">lt</span> <span class="op">=</span> <span class="fu"><a href="../reference/getBioMartGOGeneSets.html">getBioMartGOGeneSets</a></span><span class="op">(</span><span class="fl">10090</span><span class="op">)</span></span></code></pre></div>
<p>In <code><a href="../reference/getBioMartGOGeneSets.html">getBioMartGOGeneSets()</a></code>, argument <code>as_table</code> can be set to <code>TRUE</code>, then the function returns a data frame.</p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tb</span> <span class="op">=</span> <span class="fu"><a href="../reference/getBioMartGOGeneSets.html">getBioMartGOGeneSets</a></span><span class="op">(</span><span class="st">"mmusculus_gene_ensembl"</span>, as_table <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">tb</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##   go_geneset       ensembl_gene</span></span>
<span><span class="co">## 1 GO:0000002 ENSMUSG00000019699</span></span>
<span><span class="co">## 2 GO:0000002 ENSMUSG00000028982</span></span>
<span><span class="co">## 3 GO:0000002 ENSMUSG00000022615</span></span>
<span><span class="co">## 4 GO:0000002 ENSMUSG00000039264</span></span>
<span><span class="co">## 5 GO:0000002 ENSMUSG00000030557</span></span>
<span><span class="co">## 6 GO:0000002 ENSMUSG00000030879</span></span></code></pre>
<p>Argument <code>ontology</code> controls which category of GO gene sets. Possible values should be <code>"BP"</code>, <code>"CC"</code> and <code>"MF"</code>.</p>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/getBioMartGOGeneSets.html">getBioMartGOGeneSets</a></span><span class="op">(</span><span class="st">"mmusculus_gene_ensembl"</span>, ontology <span class="op">=</span> <span class="st">"BP"</span><span class="op">)</span> <span class="co"># the default one</span></span>
<span><span class="fu"><a href="../reference/getBioMartGOGeneSets.html">getBioMartGOGeneSets</a></span><span class="op">(</span><span class="st">"mmusculus_gene_ensembl"</span>, ontology <span class="op">=</span> <span class="st">"CC"</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/getBioMartGOGeneSets.html">getBioMartGOGeneSets</a></span><span class="op">(</span><span class="st">"mmusculus_gene_ensembl"</span>, ontology <span class="op">=</span> <span class="st">"MF"</span><span class="op">)</span></span></code></pre></div>
<p>Last, argument <code>gene_id_type</code> can be set to <code>"entrez_gene"</code> or <code>"gene_symbol"</code>, then genes in the gene sets are in Entrez IDs or gene symbols. Note this depends on specific organisms, that not every organism supports Entrez IDs or gene symbols.</p>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">lt</span> <span class="op">=</span> <span class="fu"><a href="../reference/getBioMartGOGeneSets.html">getBioMartGOGeneSets</a></span><span class="op">(</span><span class="st">"mmusculus_gene_ensembl"</span>, gene_id_type <span class="op">=</span> <span class="st">"entrez_gene"</span><span class="op">)</span></span>
<span><span class="va">lt</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">## $`GO:0000002`</span></span>
<span><span class="co">##  [1] "23797"  "70556"  "72962"  "83408"  "17258"  "27397"  "192287" "74143" </span></span>
<span><span class="co">##  [9] "27393"  "208084" "27395"  "17527"  "74528"  "327762" "408022" "382985"</span></span>
<span><span class="co">## [17] "83945"  "18975"  "226153" "381760" "50776"  "20133"  "11545"  "230784"</span></span>
<span><span class="co">## [25] "21975"  "12628"  "13804"  "72170"  "22059"  "76781"  "66592"  "74244" </span></span>
<span><span class="co">## [33] "16882"  "216021" "216805" "19017"</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="version-of-the-data">Version of the data<a class="anchor" aria-label="anchor" href="#version-of-the-data"></a>
</h2>
<p>The object <code>BioMartGOGeneSets</code> contains the version and source of data.</p>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">BioMartGOGeneSets</span></span></code></pre></div>
<pre><code><span><span class="co">## BioMart Gene Ontology gene sets</span></span>
<span><span class="co">##   Source: https://www.ensembl.org/info/data/biomart/index.html </span></span>
<span><span class="co">##   Number of organisms: 670 </span></span>
<span><span class="co">##   Marts: fungi_mart, genes_mart, metazoa_mart, plants_mart, protists_mart </span></span>
<span><span class="co">##   Built date: 2023-02-11, with biomaRt (2.54.0), GO.db (3.16.0)</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="session-info">Session Info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## R version 4.2.0 (2022-04-22)</span></span>
<span><span class="co">## Platform: x86_64-apple-darwin17.0 (64-bit)</span></span>
<span><span class="co">## Running under: macOS Big Sur/Monterey 10.16</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS:   /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRblas.0.dylib</span></span>
<span><span class="co">## LAPACK: /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRlapack.dylib</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">## [1] C/UTF-8/C/C/C/C</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">## [1] BioMartGOGeneSets_0.99.9 knitr_1.42              </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##  [1] bslib_0.4.2            compiler_4.2.0         jquerylib_0.1.4       </span></span>
<span><span class="co">##  [4] GenomeInfoDb_1.34.9    XVector_0.38.0         zlibbioc_1.44.0       </span></span>
<span><span class="co">##  [7] bitops_1.0-7           tools_4.2.0            digest_0.6.31         </span></span>
<span><span class="co">## [10] jsonlite_1.8.4         evaluate_0.20          memoise_2.0.1         </span></span>
<span><span class="co">## [13] lifecycle_1.0.3        rlang_1.0.6            cli_3.6.0             </span></span>
<span><span class="co">## [16] yaml_2.3.7             pkgdown_2.0.7          xfun_0.37             </span></span>
<span><span class="co">## [19] fastmap_1.1.0          GenomeInfoDbData_1.2.9 stringr_1.5.0         </span></span>
<span><span class="co">## [22] desc_1.4.2             fs_1.6.1               vctrs_0.5.2           </span></span>
<span><span class="co">## [25] sass_0.4.5             S4Vectors_0.36.1       systemfonts_1.0.4     </span></span>
<span><span class="co">## [28] IRanges_2.32.0         stats4_4.2.0           rprojroot_2.0.3       </span></span>
<span><span class="co">## [31] glue_1.6.2             R6_2.5.1               textshaping_0.3.6     </span></span>
<span><span class="co">## [34] rmarkdown_2.20         purrr_1.0.1            magrittr_2.0.3        </span></span>
<span><span class="co">## [37] htmltools_0.5.4        GenomicRanges_1.50.2   BiocGenerics_0.44.0   </span></span>
<span><span class="co">## [40] ragg_1.2.5             stringi_1.7.12         RCurl_1.98-1.10       </span></span>
<span><span class="co">## [43] cachem_1.0.6</span></span></code></pre>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Zuguang Gu.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
